pandas/core/internals.py

```diff
@@ -4066,6 +4076,27 @@ def canonicalize(block):
         return all(block.equals(oblock)
                    for block, oblock in zip(self_blocks, other_blocks))
 
+    def unstack(self, unstacker):
+        """Return blockmanager with all blocks unstacked"""
+        dummy = unstacker(np.empty((0, 0)), value_columns=self.items)
+        new_columns = dummy.get_new_columns()
+        new_index = dummy.get_new_index()
+        new_blocks = []
+        mask_columns = np.zeros_like(new_columns, dtype=bool)
+
+        for blk in self.blocks:
+            bunstacker = unstacker(
```

Comment:
I think I managed something,  but I wouldn't say it's much better now. It's still couples block unstacking with unstacked BlockManager's `new_columns`, and I'm not really sure how to get rid of that.